src/kepler_model/estimate/model_server_connector.py

```diff
@@ -8,22 +8,31 @@
 from kepler_model.util.config import is_model_server_enabled, get_model_server_req_endpoint, get_model_server_list_endpoint, download_path
 from kepler_model.util.loader import get_download_output_path
 from kepler_model.util.train_types import ModelOutputType
+from kepler_model.train.profiler.node_type_index import discover_spec_values
 
+machine_spec_mount_path = "/etc/machine_spec"
 
-# discover_spec: determine node spec in json format (refer to NodeTypeSpec)
-def discover_spec():
-    import psutil
+# get_spec_values: determine node spec in json format (refer to NodeTypeSpec)
+def get_spec_values():
```

Comment:
```suggestion
def get_spec_values(machine_id : str|None):
```
Let the caller pass the env var, this makes the function unit testable as well .